File: numberFunc/numberFunc/Graph/ScriptTemplate/Power/CGPower.js
/**
 * @file CGPower.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGPower.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGPower extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    return Math.pow(this.inputs[0](), this.inputs[1]());
  }
}

exports.CGPower = CGPower;

File: numberFunc/numberFunc/Graph/ScriptTemplate/Power/CGPower.lua
local CGPower= CGPower or {}
CGPower.__index = CGPower

function CGPower.new()
    local self = setmetatable({}, CGPower)
    self.inputs = {}
    return self
end

function CGPower:setInput(index, func)
    self.inputs[index] = func
end

function CGPower:getOutput(index)
    return math.pow(self.inputs[0](), self.inputs[1]())
end

return CGPower
